src/components/CampaignDetails.vue

```diff
@@ -58,21 +58,32 @@ export default {
         from: this.$store.state.accountNum,
         value: web3.utils.toWei(this.contribution, "ether")
       })
-      if (result && (web3.utils.toWei(this.contribution, "ether") > this.campaign.min_contribution.toString())) {
+      if (result) {
         try {
           await axios.put(
             `${VUE_APP_API_URL}campaigns/${this.campaign.id}`,
-            { contributors: (this.campaign.contributors + 1) }
+            {
+              value: (this.$store.getters.getSingleCampaign(this.address).value + parseFloat(this.contribution))
+            }
           );
         } catch (error) {
           return { error };
         }
+        if (this.contribution > this.campaign.min_contribution) {
+          try {
+            const resp = await axios.post(
+              `${VUE_APP_API_URL}campaigns/${this.campaign.address}/contributor/${this.$store.state.accountNum}`,
+              {
+                address: this.$store.state.accountNum,
+                email: null
+              }
+            );
+          } catch (error) {
```

Comment:
This is all super ugly and should be refactored if we have time, but it seems to be working!

Reply:
I think once we start moving some of this logic away of the component and into the store some of this code will be simplified.